sktime/forecasting/compose/_pipeline.py

```diff
@@ -1184,7 +1184,7 @@ class ForecastX(BaseForecaster):
     ----------
     forecaster_y : BaseForecaster
         sktime forecaster to use for endogeneous data `y`
-    forecaster_X : BaseForecaster, optional, default = forecaster_y
```

Comment:
oh, I see it was not optional after all.

Should we instead not make the change to `__init__`? It will break nothing downstream.

Out of scope though if you disagree, as you are appropriately correcting a wrong docstring.

Reply:
I'll prefer to keep `__init__` as is. I know `sktime` don't use it, but since `python3.8` there are positional-only and keyword-only arguments. I'm not sure if making `forecaster_X` optional now by switching from positional to keyword will break users' workflows or not.

Reply:
how could this break sth? I don't get the scenario